src/library.js

```diff
@@ -3252,27 +3252,20 @@ mergeInto(LibraryManager.library, {
     assert(getWasmTableEntry(ptr), 'missing table entry in dynCall: ' + ptr);
 #endif
 #if MEMORY64
-    // With MEMORY64 we have an additional step to covert `p` arguments to
+    // With MEMORY64 we have an additional step to convert `p` arguments to
     // bigint. This is the runtime equivalent of the wrappers we create for wasm
     // exports in `emscripten.py:create_wasm64_wrappers`.
-    if (sig.includes('p')) {
-      var new_args = [];
-      args.forEach((arg, index) => {
-        if (sig[index + 1] == 'p') {
-          arg = BigInt(arg);
-        }
-        new_args.push(arg);
-      });
-      args = new_args;
+    for(let i = 0; i < sig.length; ++i) {
+      if (sig[i] == 'p') args[i] = BigInt(args[i]);
     }
 #endif
-    var rtn = getWasmTableEntry(ptr).apply(null, args);
+    let rtn = getWasmTableEntry(ptr)(...args);
 #if MEMORY64
-    if (sig[0] == 'p') {
-      rtn = Number(rtn);
-    }
-#endif
+    return sig[0] == 'p' ? Number(rtn) : rtn;
```

Comment:
Does this actually result in smaller code, even after closure is applied?

Reply:
I did not test. To me this form read better, but also I think it would not be good to have a rule that we should not apply optimizations that Closure is able to do, because not everyone is able to use Closure. (e.g. we currently have challenges to adopt Closure due to it breaking many Unity asset packages that Unity game developers have created). Of course leaving the heaviest optimizations for Closure makes sense, but where it is straightforward to do by hand, I think we wouldn't want to shy away from.